docs/source/notes/autograd.rst

```diff
@@ -210,3 +210,102 @@ No thread safety on C++ hooks
+convention for autograd for Complex Numbers.
+
+Suppose we have a function :math:`F: C -> C` which we can decompose into functions u and v
+which compute the real and imaginary parts of the function:
+
+    .. code::
+
+        def F(z):
+            x, y = real(z), imag(z)
+            return u(x, y) + v(x, y) * 1j
+
+The Jacobian, JVP and VJP for function :math:`F` at :math:`(x, y)` are defined as:
+
+    .. code::
+
+        def JVP(tangent):
+            c, d = real(tangent), imag(tangent)
+            return [1, i]^T * J * [c, d]
+
+    .. code::
+
+        def VJP(cotangent):
+            c, d = real(cotangent), imag(cotangent)
+            return [c, -d]^T * J * [1, -i]
+
+    where
+
+    .. math::
+
+        J = \begin{bmatrix}
+            \partial_0u(x, y) & \partial_1u(x, y)\\
+            \partial_0v(x, y) & \partial_1v(x, y) \end{bmatrix} \\
+
+In PyTorch, the VJP is mostly what we care about, as it is the computation performed when we do backward
+mode automatic differentiation. Notice that d and i (imaginary number) are negated in the formula above.
+
+Why is there a negative sign in the formula above?
+**************************************************
+
+For a function F: V → W, where are V and W are vector spaces. The output of
+the Vector-Jacobian Product :math:`VJP : V → (W^* → V^*)` is a linear map
+from :math:`W^* → V^*` (explained in `Chapter 4 of Dougal’s thesis <https://dougalmaclaurin.com/phd-thesis.pdf>`_).
+
+The negative signs in the above VJP computation are due to conjugation. The first
+vector in the output returned by VJP for a given cotangent is a covector (\in :math:`C^*`),
+and the last vector in the output is used to get the result in :math:`C`
+since the final result of reverse-mode differentiation of a function is a covector belonging
+to :math:`C^*` (explained in `Chapter 4 of Dougal’s thesis <https://dougalmaclaurin.com/phd-thesis.pdf>`_).
+
+What happens if I call backward() on a complex scalar?
+******************************************************
+
+For geneneral ℂ→ℂ functions, the Jacobian has 4 real-valued degrees of freedom (as in the 2x2 Jacobian matrices above),
+so we can’t hope to represent all of them with in a complex number.
+
+    1. For holomorphic functions, the gradient can be fully represented with complex numbers due to the Cauchy-Riemann equations. And so,
+       we can obtain that gradient using backward which is just a call to vjp with covector 1.0.
+    2. For non-holomorphic functions, the gradient can't be fully represented with complex numbers.
+       The partial derivatives of the imaginary part of the function (:math:`v(x, y)` above) are discarded
+       (e.g., this is equivalent to dropping the imaginary part of the loss before performing a backwards).
+       For any other desired behavior, you can specify the covector :math:`v` in `torch.autograd.functional.vjp` call.
```

Comment:
actually you can specify it in the `grad_output=` parameter of `autograd.backward()` or `autograd.grad()` as well.

Reply:
oh okay gotcha. will change it to that then

Reply:
Do you plan to leave both here? Or just `autograd.grad()`, or just `.backward()`?